src/config/parser.rs

```diff
@@ -48,6 +49,9 @@ pub struct Config {
     pub tcp_connection_keep_alive: u8,
     /// It stores the pool idle connection timeout in seconds.
     pub pool_idle_connection_timeout: u8,
+
+    /// Url of the proxy to use for outgoing requests.
+    pub proxy: Option<Proxy>,
```

Comment:
```suggestion
    #[cfg(feature = "socks")]
    /// Url of the proxy to use for outgoing requests.
    pub proxy: Option<Proxy>,
```

I would suggest having this behind a `cfg` feature macro as this will only add the following code to the binary if the `socks` feature has been enabled during build otherwise it will not include it thus reducing binary size and removing unneeded code. 🙂